src/main/webapp/js/utils/table-utils.jsx

```diff
@@ -58,14 +59,67 @@ const getTableCell = (item, props) => {
 
 const getTableColumn = (props) => {
   const {
-    name, displayName, filterable,
+    name, displayName, filterable, format, fieldType,
   } = props;
 
+  const optionLabels = [];
+  if (format) {
+    _.forEach(format.split(','), (val) => {
+      const chunks = val.split(':');
+      optionLabels.push(chunks[1]);
+    });
+  }
+
   return {
     Header: displayName,
     accessor: name,
     filterable,
     Cell: item => getTableCell(item, props),
+    filterMethod: (filter, row) => {
+      let filterValue = null;
+      if (fieldType !== 'ENUM') {
+        filterValue = row[filter.id].startsWith(filter.value);
+      }
+      if (filter.value === 'all') {
+        filterValue = true;
+      }
+      if (format) {
+        _.forEach(format.split(','), (val) => {
+          const chunks = val.split(':');
+          if (filter.value === chunks[1]) {
+            filterValue = row[filter.id] === chunks[0];
+          }
+        });
+      }
+      return filterValue;
+    },
+    Filter: ({ filter, onChange }) => {
+      let button = null;
+      if (fieldType === 'ENUM') {
```

Comment:
create a separate component for the filter (it can be useful for the reports too) and add a method getTableFilter (like getTableCell)